obspy/clients/fdsn/client.py

```diff
@@ -555,6 +555,26 @@ def get_events(self, starttime=None, endtime=None, minlatitude=None,
             data_stream.close()
             return cat
 
+    def get_availability(self, network=None, station=None, location=None, 
+                         channel=None, starttime=None, endtime=None, 
+                         quality=None, minimumlength=None, longestonly=None, 
+                         format=None, filename=None, **kwargs):
+        if "availability" not in self.services:
+            msg = "The current client does not have an availability service."
+            raise ValueError(msg)
+
+        locs = locals()
+        setup_query_dict('availability', locs, kwargs)
+
+        url = self._create_url_from_parameters(
+            "availability", DEFAULT_PARAMETERS['availability'], kwargs)
+
+        availability = self._download(url, return_string=True).decode()
+        lines = [line.split() for line in availability.strip().split('\n')[1:]] # skip header 
```

Comment:
yes, you are right

Reply:
actually, what I have done is to change the call to Client._download() so that it gets a BytesIO object instead of a string, then get the lines with BytesIO.readlines(). 

The filename parameter is not handled yet. get_stations, get_events, get_waveforms all permit writing to file; get_availability should too?

Reply:
What I have done is to delete the `return_string=True` parameterization in the `Client._download` call so that it returns a BytesIO object instead of a string. Then read the lines with `BytesIO.readlines`. I think this seems consistent with the approach in the other get_* methods.